### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by FlorinPeter**
# Proposed Changes

- added Dockerfile
- added multiPath route
- added prefer network
- added Unsubscribe
- fix nil pointer dereference
- fix error message

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM golang:1.19-bullseye AS builder
+
+WORKDIR /workspace
+COPY . .
```
Comment (afritzler) [inline Dockerfile:4]: I would not copy all the content but just the relevant files.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+RUN make amd64
+
+FROM debian:bookworm
```
Comment (afritzler) [inline Dockerfile:7]: I would recommend using the `slim` version as a base image.

File: netlink.go [hunk 3]
```diff
@@ -109,21 +147,62 @@ func (c *NetlinkClient) RemoveRoute(vni VNI, dest Destination, hop NextHop) erro
...
 	} // by default, the route is already installed into the kernel table without explicite specification
 
-	if err := netlink.RouteDel(route); err != nil {
-		return fmt.Errorf("cannot remove route to %s (table %d) from kernel: %v", dest, table, err)
+	err, _ = c.routeTable.RemoveNextHop(vni, dest, hop, c.mbp)
+	if err != nil {
+		return fmt.Errorf("cannot add route to internal table vni: %s dest: %s hop: %s error: %v", vni, dest, hop, err)
+	}
+
+	if c.config.PreferNetwork != nil {
```
Comment (afritzler) [inline netlink.go:161]: This code looks very similar to the one above. Can we move this into a helper method?

**Review by afritzler — CHANGES_REQUESTED** [review]
Some small changes. In general would like to see some test cases in the `metalbond` project.

Maybe @byteocean can also have a look at this PR as I can not judge how the route table configuration will affect our current setup.

**Comment by MalteJ** [comment]
puh, that's a huge PR touching a lot of things.

What's preferNetwork?

**Comment by MalteJ** [comment]
Sorry, this PR is too big, I cannot understand it.

**Comment by FlorinPeter** [comment]
@MalteJ then let's go together thru the changes

**Pull request closed without merging**